tests/integ/runner/test_integ_runner.py

```diff
@@ -244,6 +244,74 @@ def test_exec_compliance_stateless_aws_verifiedpermissions_policy(
         assert warning_result == compliance_result.warning[warning_rule]
 
 
+@pytest.mark.parametrize(
+    "collected_schemas,collected_rules,non_compliant_rules,warning_rules",
+    [
+        (
+            collect_schemas(
+                schemas=[
+                    "file:/"
+                    + str(
+                        Path(os.path.dirname(os.path.realpath(__file__))).joinpath(
+                            "../data/schema-no-tagging-permission.json"
+                        )
+                    )
+                ]
+            ),
+            [],
+            {},
+            {
+                "ensure_arn_properties_type_string": {
+                    GuardRuleResult(
+                        check_id="ARN001",
+                        message="arn related property MUST have pattern specified",
+                        path="/properties/ManagedPolicyArns/type",
+                    )
+                },
+                "ensure_arn_properties_contain_pattern": {
+                    GuardRuleResult(
+                        check_id="ARN002",
+                        message="arn related property MUST have pattern specified",
+                        path="",
+                    )
+                },
+                "ensure_properties_do_not_support_multitype": {
+                    GuardRuleResult(
+                        check_id="COM002",
+                        message="type MUST NOT have combined definition",
+                        path="/properties/AssumeRolePolicyDocument/type",
+                    )
+                },
```

Comment:
not necessary, your test iterate over `non_compliant_rules,warning_rules` which are parameters, so you dont check every single check that fails

Reply:
Updated to only put the TAG012 now~